packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -949,19 +943,31 @@ class VM extends ServiceObjectOwner {
     return invokeRpcRaw('_getVMTimeline', timeout: kLongRequestTimeout);
   }
 
-  Future<void> refreshViews() {
+  Future<void> refreshViews({ bool waitForViews = false }) async {
+    assert(waitForViews != null);
+    assert(loaded);
     if (!isFlutterEngine)
-      return Future<void>.value(null);
-    _viewCache.clear();
-    final Completer<void> completer = Completer<void>();
-    final List<Future<ServiceObject>> futures = <Future<ServiceObject>>[];
-    for (Isolate isolate in isolates.toList()) {
-      futures.add(vmService.vm.invokeRpc<ServiceObject>('_flutter.listViews',
+      return;
+    int failCount = 0;
+    while (true) {
+      _viewCache.clear();
+      final List<Future<void>> futures = <Future<void>>[];
+      for (Isolate isolate in isolates.toList()) {
+        futures.add(vmService.vm.invokeRpc<ServiceObject>(
+          '_flutter.listViews',
           timeout: kLongRequestTimeout,
-          params: <String, dynamic> {'isolateId': isolate.id}));
+          params: <String, dynamic> {'isolateId': isolate.id},
+        ));
+      }
+      await Future.wait(futures);
+      if (_viewCache.values.isNotEmpty || !waitForViews)
```

Comment:
Can `listViews` ever legitimately turn up no `FlutterView` instances?  If so, we'll retry here until we eventually fail, even though there was no failure...

Reply:
Though I guess that's what `waitForViews` is for (why why it defaults to `false`)?

Reply:
I don't know the answer to your first question, but if so, I don't know how we would distinguish that case from the normal case where we don't want to return until we have a view.

The argument defaults to false so as to not change the default behaviour in the regular case; only in the specific case where I believe we definitely need a view did I set it to true.